pkg/api/pod/warnings.go

```diff
@@ -255,6 +255,34 @@ func warningsForPodSpecAndMeta(fieldPath *field.Path, podSpec *api.PodSpec, meta
 		return true
 	})
 
+	// Accumulate ports across all containers
+	allPorts := map[string]portBlock{}
+	pods.VisitContainersWithPath(podSpec, fieldPath.Child("spec"), func(c *api.Container, fldPath *field.Path) bool {
+		for i, port := range c.Ports {
+			k := fmt.Sprintf("%d/%s", port.ContainerPort, port.Protocol)
+			if other, found := allPorts[k]; found {
+				// Someone else has this protcol+port, but it still might not be a conflict.
+				conflict := false
+
+				// Exact matches are obvious.
+				if port.HostIP == other.port.HostIP && port.HostPort == other.port.HostPort {
+					conflict = true
+				}
+				// Unspecified fields are more subtle.
+				if port.HostPort == 0 || other.port.HostPort == 0 || port.HostIP == "" || other.port.HostIP == "" {
```

Comment:
Is there any example on when we're entering 0 as the input?

Reply:
https://github.com/kubernetes/kubernetes/blob/f6416ebf266a17c5e8c862e1025fb488dfc4a388/staging/src/k8s.io/api/core/v1/types.go#L1929-L1934

hostPort cannot be 0, so I may update the code here.

Reply:
I also see there is another comment that hostPort 0 means random port.